instrumentation/opentelemetry-instrumentation-asyncio/src/opentelemetry/instrumentation/asyncio/__init__.py

```diff
@@ -254,7 +254,9 @@ def trace_item(self, coro_or_future):
         # Task is already traced, return it
         if isinstance(coro_or_future, asyncio.Task):
             return coro_or_future
-        if asyncio.iscoroutine(coro_or_future):
+        if asyncio.iscoroutine(coro_or_future) and hasattr(
+            coro_or_future, "__name__"
```

Comment:
Would it make more sense to move this check into the `trace_coroutine` as an early return? I did some digging and `__name__` is not part of the Coroutine ABC (I think may only be defined on the native coroutine type)